### PR TITLE
Move fork module to rollup interface

### PR DESCRIPTION
**Pull request by rakanalh**
# Description
Citrea primitives crate should not be imported by anything sov-sdk related. This would later lead to cyclic dependencies.

### CODE REVIEW

**Review by eyusufatik — APPROVED** [review]

File: bin/citrea/provers/risc0/guest-bitcoin/Cargo.lock [hunk 1]
```diff
@@ -2919,6 +2919,7 @@ dependencies = [
  "serde",
  "sha2",
  "thiserror",
+ "tracing",
```
Comment (kpp) [inline bin/citrea/provers/risc0/guest-bitcoin/Cargo.lock:2922]: Wow

Comment (rakanalh) [reply bin/citrea/provers/risc0/guest-bitcoin/Cargo.lock:2922]: The dependency is optional and would only be used in native mode

Comment (eyusufatik) [reply bin/citrea/provers/risc0/guest-bitcoin/Cargo.lock:2922]: then it shouldn't have ended up in the bitcoin prover cargo.lock

File: crates/sovereign-sdk/rollup-interface/Cargo.toml [hunk 2]
```diff
@@ -43,14 +44,15 @@ fuzzing = ["proptest", "proptest-derive", "sha2", "std"]
...
+  "borsh/default",
+  "borsh/bytes",
+  "bytes",
+  "digest/default",
+  "hex/default",
+  "proptest?/default",
+  "serde/default",
+  "sha2?/default",
+  "thiserror",
+  "tracing",
```
Comment (eyusufatik) [inline crates/sovereign-sdk/rollup-interface/Cargo.toml:57]: only put in native and it'll be solved

Comment (eyusufatik) [reply crates/sovereign-sdk/rollup-interface/Cargo.toml:57]: I mean tracing

Comment (rakanalh) [reply crates/sovereign-sdk/rollup-interface/Cargo.toml:57]: i did

Comment (rakanalh) [reply crates/sovereign-sdk/rollup-interface/Cargo.toml:57]: Ahh i see what you mean... it's in std so it should also be included with native

Comment (eyusufatik) [reply crates/sovereign-sdk/rollup-interface/Cargo.toml:57]: no its in std which is in native. I believe we should put it explicitly in native, like tokio

**Pull request merged**